### PR TITLE
Add support for using the WebRTC framework for capturing audio.

### PR DESCRIPTION
**Pull request by oranoceallaigh**

### CODE REVIEW

File: src/js/voysis.js [hunk 1]
```diff
@@ -262,20 +238,161 @@
...
+        }
+        this.mediaRecorder.ondataavailable = function(evt) {
+            var audioDataBlob = evt.data;
+            if (args_.saveAudioStream) {
+                savedAudioStream_[savedAudioStream_.length] = audioDataBlob;
+            }
+            onDataAvailable(audioDataBlob);
+        };
+        this.mediaRecorder.onerror = errorHandler;
+        this.mediaRecorder.start(128);
```
Comment (ttroy50) [inline src/js/voysis.js]: what does this 128 mean?

File: src/js/voysis.js [hunk 2]
```diff
@@ -262,20 +238,161 @@
...
+     *     // @return The audio stream blob, or null if no audio data is available
+     *     //         or the recorder doesn't support saving audio locally.
+     *     Blob getSavedAudioSteam();
+     * }
+     */
+    function WebRtcRecorder() {
+    }
+    WebRtcRecorder.implName = 'WebRtcRecorder';
+    WebRtcRecorder.prototype.getMimeType = function() {
+        return 'audio/webm';
```
Comment (ttroy50) [inline src/js/voysis.js]: Are we not going to send the information on sample rate, encoding type in the mimetype header for webm?

File: src/js/voysis.js (hunk 1 above)
Comment (oranoceallaigh) [reply src/js/voysis.js]: The mediarecorder start method accepts a "time slice" parameter, it's the number of milliseconds to fill in each blob delivered to `ondataavailable`. Without this, we'd have to set up our own browser timer and call `requestData`.

The number of samples/bytes generated per timeslice is going to be dependent on the sample rate, which I do not (and cannot) know. WebRTC is designed to use vorbis or opus for audio streaming and therefore deliver in bits-per-second. Chrome happens to support using a PCM codec but I don't know what sample rate it's going to be delivered at. I therefore have to pick a reasonable value that'll give a managable number of bytes at 16KHz, 44k1Hz and 48KHz.

File: src/js/voysis.js (hunk 2 above)
Comment (oranoceallaigh) [reply src/js/voysis.js]: No, this impl is similar to us sending an `audio/wav` stream. Anyway, I don't have any of the information to populate - all I know is I've requested an `audio/webm` stream from WebRTC - that framework will encode the details of the audio within the webm data itself. We parse that out server side.

Comment (ttroy50) [reply src/js/voysis.js]: I figured that's what you would say. Do you parse it out on the server side so that we can easily query it later?

Comment (oranoceallaigh) [reply src/js/voysis.js]: Yep, it's stored in a queryable form

**Pull request merged**